### ISSUE TITLE
ksort() behavior changed since PHP 8.2

### ISSUE THREAD

**Issue by mabar** [php/doc-en#3307]
`ksort()` before PHP 8.2 sorted first string keys, then numeric keys. Since PHP 8.2 it sorts first numeric keys, then string keys.

I think this change should be noted in `ksort` documentation and in the changelog.

I have tested these flags:
- `SORT_REGULAR`, `SORT_ASC`, `SORT_DESC` - behavior changed
- `SORT_NATURAL` - behavior is the same

I don't know if any other sorting functions are affected

See https://3v4l.org/T2EDs

**Comment by cmb69** [comment]
Indeed, so far this is only mentioned in the [migration guide](https://www.php.net/manual/en/migration82.incompatible.php#migration82.incompatible.standard),
